### PR TITLE
Rewrite json tests based on criterion

### PR DESCRIPTION
**Pull request by norberttak**
rewrite test_json_parser, test_dot_notation based on criterion

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: modules/json/tests/test_json_parser.c [hunk 1]
```diff
@@ -71,76 +46,91 @@ parse_json_into_log_message_no_check(const gchar *json)
...
+  app_shutdown();
+}
+
+Test(json_parser, test_json_parser_parses_well_formed_json_and_puts_results_in_message, .init = setup, .fini = teardown)
 {
   LogMessage *msg;
+  LogParser *json_parser = json_parser_new(NULL);
 
-  msg = parse_json_into_log_message("{'foo': 'bar'}");
+  msg = parse_json_into_log_message("{'foo': 'bar'}", json_parser);
```
Comment (Kokan) [inline modules/json/tests/test_json_parser.c]: I think you missed here the `log_pipe_unref` for `json_parser`.

File: modules/json/tests/test_dot_notation.c [hunk 2]
```diff
@@ -93,53 +72,59 @@ assert_dot_notation_eval_fails(const gchar *input_json, const gchar *subscript)
...
+}
+
+Test(json, test_dot_notation_eval_empty_subscript_returns_the_object, .init = setup, .fini = teardown)
 {
   assert_dot_notation_eval_equals("{'foo': 'bar'}", "", "{'foo': 'bar'}");
 }
 
-static void
-test_dot_notation_eval_fails_with_an_identifier_that_doesnt_start_with_a_letter(void)
+Test(json, test_dot_notation_eval_fails_with_an_identifier_that_doesnt_start_with_a_letter, .init = setup,
```
Comment (Kokan) [inline modules/json/tests/test_dot_notation.c]: You could define a `TestSuite`, in which case you do not have to repeat the `.init = setup, .fini = teardown` part.

File: modules/json/tests/test_json_parser.c [hunk 3]
```diff
@@ -151,61 +141,41 @@ test_json_parser_validate_type_representation(void)
...
-  msg = parse_json_into_log_message("[{'foo':'bar'}, {'bar':'foo'}]");
+  msg = parse_json_into_log_message("[{'foo':'bar'}, {'bar':'foo'}]", json_parser);
   assert_log_message_value(msg, log_msg_get_value_handle("foo"), "bar");
   log_msg_unref(msg);
+  log_pipe_unref(&json_parser->super);
 }
 
-static void
-test_json_parser_works_with_templates(void)
+Test(json_parser, test_json_parser_works_with_templates, .init = setup, .fini = teardown)
```
Comment (Kokan) [inline modules/json/tests/test_json_parser.c]: You could define a `TestSuite`, in which case you do not have to repeat the `.init = setup, .fini = teardown` part.

File: modules/json/tests/test_dot_notation.c (hunk 2 above)
Comment (norberttak) [reply modules/json/tests/test_dot_notation.c]: fixed

File: modules/json/tests/test_json_parser.c (hunk 3 above)
Comment (norberttak) [reply modules/json/tests/test_json_parser.c]: fixed

**Pull request merged**